### PR TITLE
Correct `FunctionNode.forward` output type message

### LINKED ISSUE #7654: Confusing error message about `FunctionNode.forward` output type
Code:
```python
import chainer, numpy

class MyFunc(chainer.FunctionNode):
    def forward_cpu(self, inputs):
        x, = inputs
        y = numpy.square(x),
        return y,  # nested tuple

x = chainer.Variable(numpy.ones((2,), 'float32'))
MyFunc().apply((x,))
```

Error message:
```
TypeError: incompatible array types are mixed in the forward output (MyFunc).
Actual: <class 'tuple'>
```

The message does not correctly describe the mistake.

### PR DESCRIPTION
**Pull request by niboshi**
Fixes #7654

### CODE REVIEW

**Comment by kmaehashi** [comment]
pfnCI, test this please.

File: tests/chainer_tests/test_function_node.py [hunk 1]
```diff
@@ -426,14 +426,27 @@ def forward(self, inputs):
...
+    def test_invalid_output_type(self):
+        class FunctionNode(chainer.FunctionNode):
+
+            def forward(self, inputs):
+                return object(),
+
+        f = FunctionNode()
+        x1 = chainer.Variable(self.x1)
+
+        with self.assertRaisesRegex(
```
Comment (kmaehashi) [inline tests/chainer_tests/test_function_node.py]: Should use `six.assertRaisesRegex(self, ...)` for Py2 compatibility.

Comment (niboshi) [reply tests/chainer_tests/test_function_node.py]: Thanks, fixed as suggested.

**Comment by niboshi** [comment]
PTAL

**Comment by niboshi** [comment]
ping

**Comment by toslunar** [comment]
Jenkins & flexCI, test this please.

**Comment by toslunar** [comment]
Jenkins, test this please.

**Comment by kmaehashi** [comment]
LGTM!
pfnCI, test this please.

**Comment by toslunar** [comment]
Jenkins, test this please.

**Pull request merged**